periodic sync upstream KF to midstream ODH

tarilabs opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
<!--- [UI] Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

If you have UI changes

<!--- You can ignore these if you are doing Go Model Registry REST server, MR Python client, manifest, controller, internal logic, etc changes; aka non-UI / visual changes -->
- [ ] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

superseeds https://github.com/opendatahub-io/model-registry/pull/154

applying labels

lampajr commented:
`Test container image build` fails because it uses Dockerfile from `main` branch (which obviously does not contain the fix) - we should merge this https://github.com/opendatahub-io/model-registry/pull/153 first to get the proper result here.

tarilabs reviewed (approved):
Thank you for having kept this in a single PR(/commit) when we squash and merge, I would recommend noting the squash commit in the GitHub issue just for completeness, this way one day we will need to restore this code we could do with reverting the commit specifically minor likely conflicts just for Dockerfile, Makeffile, go.mod and go.sum

lampajr commented:
> Thank you for having kept this in a single PR(/commit) when we squash and merge, I would recommend noting the squash commit in the GitHub issue just for completeness, this way one day we will need to restore this code we could do with reverting the commit specifically minor likely conflicts just for Dockerfile, Makeffile, go.mod and go.sum

+1 I agree

> `Test container image build` fails because it uses Dockerfile from `main` branch (which obviously does not contain the fix) - we should merge this #153 first to get the proper result here.

Fixed on `main` branch (https://github.com/opendatahub-io/model-registry/pull/153 merged), `Test container image build` working now and green.

tarilabs committed:
chore: bump py version (#618)

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

dependabot[bot] committed:
build(deps-dev): bump @cypress/code-coverage from 3.13.7 to 3.13.8 in /clients/ui/frontend (#624)

Bumps [@cypress/code-coverage](https://github.com/cypress-io/code-coverage) from 3.13.7 to 3.13.8.
- [Release notes](https://github.com/cypress-io/code-coverage/releases)
- [Changelog](https://github.com/cypress-io/code-coverage/blob/master/.releaserc)
- [Commits](https://github.com/cypress-io/code-coverage/compare/v3.13.7...v3.13.8)

---
updated-dependencies:
- dependency-name: "@cypress/code-coverage"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump @mui/icons-material from 6.1.9 to 6.1.10 in /clients/ui/frontend (#625)

Bumps [@mui/icons-material](https://github.com/mui/material-ui/tree/HEAD/packages/mui-icons-material) from 6.1.9 to 6.1.10.
- [Release notes](https://github.com/mui/material-ui/releases)
- [Changelog](https://github.com/mui/material-ui/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mui/material-ui/commits/v6.1.10/packages/mui-icons-material)

---
updated-dependencies:
- dependency-name: "@mui/icons-material"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump huggingface-hub from 0.26.3 to 0.26.5 in /clients/python (#619)

Bumps [huggingface-hub](https://github.com/huggingface/huggingface_hub) from 0.26.3 to 0.26.5.
- [Release notes](https://github.com/huggingface/huggingface_hub/releases)
- [Commits](https://github.com/huggingface/huggingface_hub/compare/v0.26.3...v0.26.5)

---
updated-dependencies:
- dependency-name: huggingface-hub
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump aiohttp from 3.11.9 to 3.11.10 in /clients/python (#620)

---
updated-dependencies:
- dependency-name: aiohttp
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump coverage from 7.6.8 to 7.6.9 in /clients/python (#621)

Bumps [coverage](https://github.com/nedbat/coveragepy) from 7.6.8 to 7.6.9.
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.6.8...7.6.9)

---
updated-dependencies:
- dependency-name: coverage
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump pydantic from 2.10.2 to 2.10.3 in /clients/python (#622)

Bumps [pydantic](https://github.com/pydantic/pydantic) from 2.10.2 to 2.10.3.
- [Release notes](https://github.com/pydantic/pydantic/releases)
- [Changelog](https://github.com/pydantic/pydantic/blob/main/HISTORY.md)
- [Commits](https://github.com/pydantic/pydantic/compare/v2.10.2...v2.10.3)

---
updated-dependencies:
- dependency-name: pydantic
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump ruff from 0.8.1 to 0.8.2 in /clients/python (#623)

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.8.1 to 0.8.2.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.8.1...0.8.2)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

lampajr committed:
feat(csi): restructure the CSI as different mr executable (#601)

* Restructure the CSI as different executable

Signed-off-by: Andrea Lamparelli <a.lamparelli95@gmail.com>

* Move csi under /cmd

Signed-off-by: Andrea Lamparelli <a.lamparelli95@gmail.com>

---------

Signed-off-by: Andrea Lamparelli <a.lamparelli95@gmail.com>

dependabot[bot] committed:
build(deps-dev): bump types-python-dateutil from 2.9.0.20241003 to 2.9.0.20241206 in /clients/python (#636)

Bumps [types-python-dateutil](https://github.com/python/typeshed) from 2.9.0.20241003 to 2.9.0.20241206.
- [Commits](https://github.com/python/typeshed/commits)

---
updated-dependencies:
- dependency-name: types-python-dateutil
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tarilabs committed:
merge sync upstream KF/model-registry to midstream ODH

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

tarilabs reviewed (commented):
- keep https://github.com/kubeflow/model-registry/pull/618
- keep https://github.com/kubeflow/model-registry/pull/624
- keep https://github.com/kubeflow/model-registry/pull/625
- keep https://github.com/kubeflow/model-registry/pull/619
- keep https://github.com/kubeflow/model-registry/pull/620
- keep https://github.com/kubeflow/model-registry/pull/621
- keep https://github.com/kubeflow/model-registry/pull/622
- keep https://github.com/kubeflow/model-registry/pull/623
- keep https://github.com/kubeflow/model-registry/pull/601
but resolve conflicts with 43171af32733a675c7f43910bb3f88238eaf0a51
- keep https://github.com/kubeflow/model-registry/pull/636
- 43171af32733a675c7f43910bb3f88238eaf0a51 resolving conflicts with merging https://github.com/kubeflow/model-registry/pull/601 into midstream

labels are applied.
